### ISSUE TITLE
Converting rooms works fine in editor but fails at runtime

### ISSUE THREAD

**Issue by timothyqiu** [godotengine/godot#51000]
### Godot version

3.4.beta2, 3.x (5d4352f)

### System information

macOS 11.4

### Issue description

Clicking "Convert Rooms" works properly in editor. But calling `rooms_convert()` fails with several warnings:

```
_second_pass_portals
WARNING: Portal_Hallway possibly in the wrong room.
     at: _check_portal_for_warnings (scene/3d/room_manager.cpp:1086)
WARNING: Portal_Hallway*0 possibly in the wrong room.
     at: _check_portal_for_warnings (scene/3d/room_manager.cpp:1086)
WARNING: Portal_Hallway*0 possibly facing the wrong way.
     at: _check_portal_for_warnings (scene/3d/room_manager.cpp:1101)
_autolink_portals
ROOM	A
		POUT	Portal_Hallway
			GEOM	Floor
			GEOM	Floor2
			GEOM	Floor3
			GEOM	Floor4
			GEOM	Floor5
WARNING: QuickHull failed building simplified bound
     at: _build_simplified_bound (scene/3d/room_manager.cpp:1514)
ROOM	Hallway
		PIN 	Portal_Hallway
		PIN 	Portal_Hallway*0
			GEOM	CSGPolygon
			GEOM	CSGPolygon2
			GEOM	CSGPolygon3
ROOM	B
		POUT	Portal_Hallway*0
			GEOM	Floor
			GEOM	Floor2
			GEOM	Floor3
			GEOM	Floor4
			GEOM	Floor5
WARNING: QuickHull failed building simplified bound
     at: _build_simplified_bound (scene/3d/room_manager.cpp:1514)
Room conversion complete. 3 rooms, 2 portals.
```

![layout](https://user-images.githubusercontent.com/372476/127463334-ec032836-99ec-434a-8353-23a45f69d385.png)

The portals are all in the square rooms facing outward. Check out the MRP.

### Steps to reproduce

Use the MRP, clicking "Convert Rooms" in editor (works fine), and run the project (errors are printed).

### Minimal reproduction project

[demo.zip](https://github.com/godotengine/godot/files/6899164/demo.zip)

**Comment by lawnjelly** [comment]
This looks like a CSG updating issue. CSG stuff I added because people like to use them to preview levels but they haven't been as tested as `MeshInstance`s, so I apologize.

I've found that changing your script to this solves it:
```
func _ready():
	call_deferred("start")
	pass
	
func start():
	$RoomManager.rooms_convert()
```

I'm presuming because the CSG's have 1 frame to calculate themselves before they are queried by the RoomManager for geometry on conversion. It is a bug though, presumably there is some method in them that I need to call to force an update for this situation. :+1:
